platform/linux-generic/odp_rwlock.c

```diff
@@ -33,9 +33,14 @@ void odp_rwlock_read_lock(odp_rwlock_t *rwlock)
 
 int odp_rwlock_read_trylock(odp_rwlock_t *rwlock)
 {
-	uint32_t zero = 0;
+	uint32_t cnt = odp_atomic_load_u32(&rwlock->cnt);
+
+	while (cnt != (uint32_t)-1) {
```

Comment:
This is still not correct. Trylock should not repeatedly try the operation. Change while to if here.